go/ql/lib/semmle/go/dataflow/SsaImpl.qll

```diff
@@ -290,6 +292,172 @@ private module Internal {
     or
     rewindReads(bb, i, v) = 1 and result = getDefReachingEndOf(bb.getImmediateDominator(), v)
   }
+
+  private module AdjacentUsesImpl {
```

Comment:
Was it not possible to use the shared SSA library?

Reply:
I got stuck trying to do that before, so since I had an almost-working implementation already I decided to get that merged and switch to the shared SSA library at some point in the future.